client/src/components/toPayItem.js

```diff
@@ -74,22 +78,25 @@ export default class ToPayItem extends React.Component {
         <div style = {ItemStyle}>
           <ListItem
             innerDivStyle ={ListItemStyle}
-            secondaryText ={this.props.message}
-            primaryText ={'$' + this.props.amount.toFixed(2)}
+            secondaryText ={this.props.message.message.message}
+            primaryText ={this.props.message.invoice.amount.currency + ' ' + this.props.message.invoice.amount.amount.toFixed(2)}
             rightIconButton={<IconButton onTouchTap={this.handleMsgButton}><CommunicationEmail/></IconButton>}
             secondaryTextLines={1}
+            onMouseDown={()=> this.props.router.push({pathname: '/viewInvoice', query:{key:this.props.keyID} })}
```

Comment:
Do we still need `query:{key:this.props.keyID}` if you already have `selectedMessage` inside the `messages`? If not, let's remove this string composed URL.